tests/src/Unit/EventSubscriber/EdgeExceptionSubscriberTest.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+/**
+ * Copyright 2019 Google Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License version 2 as published by the
+ * Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc., 51
+ * Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+namespace Drupal\apigee_edge\Unit\EventSubscriber;
+
+use Apigee\Edge\Exception\ApiException;
+use Drupal\apigee_edge\EventSubscriber\EdgeExceptionSubscriber;
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\Routing\RedirectDestinationInterface;
+use Drupal\Tests\UnitTestCase;
+use Drupal\Core\Config\Config;
+use Prophecy\Argument;
+use Psr\Log\LoggerInterface;
+use stdClass;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Event\GetResponseForExceptionEvent;
+use Symfony\Component\HttpKernel\HttpKernelInterface;
+use Symfony\Component\Routing\Matcher\UrlMatcherInterface;
+use Symfony\Component\Routing\RequestContext;
+
+/**
+ * Test EdgeExceptionSubscriber.
+ *
+ * @group apigee_edge
+ */
+class EdgeExceptionSubscriberTest extends UnitTestCase {
```

Comment:
Why not `\Drupal\KernelTests\KernelTestBase`?

Reply:
Created unit test to speed up testing.